src/features/auth/can-user-edit.ts

```diff
@@ -1,15 +1,13 @@
+import type { User } from "@prisma/client";  
+
 interface Entity {
   owner: {
     id: string;
   };
 }
 
-interface User {
-  id: string;
-}
-
 export const canUserEdit = <T extends Entity>(
-  user: User | null,
+  user: Pick<User, "id"> | null,
```

Comment:
Hmm i am not sure if we want to do this. This function is purposefully set to use a general purpose interface, and with this change we start to depend prisma for this implementation, which does not add any benefit.